contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinMooniswap.sol

```diff
@@ -0,0 +1,107 @@
+
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-erc20/contracts/src/v06/LibERC20TokenV06.sol";
+import "@0x/contracts-erc20/contracts/src/v06/IERC20TokenV06.sol";
+import "@0x/contracts-erc20/contracts/src/v06/IEtherTokenV06.sol";
+import "./MixinAdapterAddresses.sol";
+
+
+/// @dev Moooniswap pool interface.
+interface IMooniswapPool {
+
+    function swap(
+        IERC20TokenV06 sellToken,
+        IERC20TokenV06 buyToken,
+        uint256 sellAmount,
+        uint256 minBoughtAmount,
+        address referrer
+    )
+        external
+        payable
+        returns (uint256 boughtAmount);
+}
+
+/// @dev Moooniswap registry interface.
+interface IMooniswapRegistry {
+
+    function pools(
+        IERC20TokenV06 token1,
+        IERC20TokenV06 token2
+    )
+        external
+        view
+        returns (IMooniswapPool);
+}
+
+/// @dev BridgeAdapter mixin for mooniswap.
+contract MixinMooniswap is
+    MixinAdapterAddresses
+{
+    using LibERC20TokenV06 for IERC20TokenV06;
+    using LibERC20TokenV06 for IEtherTokenV06;
+
+    /// @dev Mooniswap registry contract.
+    IMooniswapRegistry private immutable REGISTRY;
+    /// @dev WETH token.
+    IEtherTokenV06 private immutable WETH;
+
+    constructor(AdapterAddresses memory addresses)
+        public
+    {
+        REGISTRY = IMooniswapRegistry(addresses.mooniswapRegistry);
+        WETH = IEtherTokenV06(addresses.weth);
+    }
+
+    function _tradeMooniswap(
+        IERC20TokenV06 buyToken,
+        uint256 sellAmount,
+        bytes memory bridgeData
+    )
+        internal
+        returns (uint256 boughtAmount)
+    {
+        (IERC20TokenV06 sellToken, IMooniswapPool pool) =
+            abi.decode(bridgeData, (IERC20TokenV06, IMooniswapPool));
```

Comment:
We now encode the pool as a second parameter in the bridge data. `MooniswapBridge` just does `abi.decode(bridgeData, (address))` but it should still work for both.